core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -894,7 +894,7 @@ object KafkaConfig {
     "record batch size accepted by the broker is defined via <code>message.max.bytes</code> (broker config) or " +
     "<code>max.message.bytes</code> (topic config)."
   val NumReplicaFetchersDoc = "Number of fetcher threads used to replicate messages from a source broker. " +
-  "Increasing this value can increase the degree of I/O parallelism in the follower broker."
+  "Increasing this value can increase the degree of I/O parallelism in the follower and leader broker at the cost of higher CPU utilization."
```

Comment:
Increasing the number of threads in a process doesn't just increase the CPU utilization but it also consumes memory. I don't think it is necessary to explain the cost of creating a thread in the documentation since the reader is assumed to be having a technical background. Hence, I don't think this change is needed.

Reply:
I agree for many users the additional explanation is unnecessary, but there is a vast range of expertise among all operators. I don't think there's any harm emphasizing the fact that there is a tradeoff here. I've seen some users increase this to 16 or higher and then act surprised when CPU usage becomes a problem.

Reply:
Sure. Can't argue against actual user anecdotes.